web/models.py

```
@@ -262,6 +262,103 @@ def average_rating(self):
+    
+    def is_valid(self):
+        now = timezone.now()
+        if not self.is_active:
+            return False
+        if self.valid_until and now > self.valid_until:
+            return False
+        if now < self.valid_from:
+            return False
+        if self.max_uses and self.uses_count >= self.max_uses:
+            return False
+        return True
```

Ruff (0.8.2)

Return the negated condition directly

Inline condition

(SIM103)